apps/mobile/src/features/HistoryAdvancedDetails/utils/formatHistoryTxDetails.tsx

```diff
@@ -0,0 +1,224 @@
+import React from 'react'
+import { TransactionDetails } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { ListTableItem } from '@/src/features/ConfirmTx/components/ListTable'
+import { Text, View } from 'tamagui'
+import { CopyButton } from '@/src/components/CopyButton'
+import { Address } from '@/src/types/address'
+import { isMultisigDetailedExecutionInfo } from '@/src/utils/transaction-guards'
+import { Operation } from '@safe-global/safe-gateway-typescript-sdk'
+import { AddressDisplay } from '@/src/components/AddressDisplay'
+import { Badge } from '@/src/components/Badge'
+import { HexDataDisplay } from '@/src/components/HexDataDisplay'
+
+interface formatHistoryTxDetailsProps {
+  txDetails?: TransactionDetails
+}
+
+export interface HistoryTxDetailsSection {
+  title?: string
+  items: ListTableItem[]
+}
+
+const formatHistoryTxDetails = ({ txDetails }: formatHistoryTxDetailsProps): HistoryTxDetailsSection[] => {
+  const sections: HistoryTxDetailsSection[] = []
+
+  if (!txDetails) {
+    return sections
+  }
+
+  // Section 1: Basic Transaction Info (Nonce, safeTxHash)
+  const basicInfoItems: ListTableItem[] = []
+
+  if (isMultisigDetailedExecutionInfo(txDetails.detailedExecutionInfo)) {
+    const executionInfo = txDetails.detailedExecutionInfo
+
+    basicInfoItems.push({
+      label: 'Nonce',
+      render: () => <Text>{executionInfo.nonce}</Text>,
+    })
+
+    // Safe Tx Hash
+    if (executionInfo.safeTxHash) {
+      basicInfoItems.push({
+        label: 'safeTxHash',
+        render: () => (
+          <AddressDisplay
+            address={executionInfo.safeTxHash as Address}
+            copyProps={{ color: '$textSecondaryLight', size: 16 }}
+            showIdenticon={false}
+            showExternalLink={false}
+          />
+        ),
+      })
+    }
+  }
+
+  if (basicInfoItems.length > 0) {
+    sections.push({
+      items: basicInfoItems,
+    })
+  }
+
+  // Section 2: Parameters
+  const parametersItems: ListTableItem[] = []
+
+  // Call
+  if (txDetails.txData?.operation !== undefined) {
+    const operationText = txDetails.txData.operation === Operation.CALL ? 'transfer' : 'delegateCall'
+    parametersItems.push({
+      label: 'Call',
+      render: () => (
+        <Badge
+          circular={false}
+          content={operationText}
+          themeName="badge_background"
+          circleProps={{ paddingHorizontal: 8, paddingVertical: 2, borderRadius: 5 }}
+        />
+      ),
+    })
+  }
+
+  // Contract (same as To in the screenshot)
+  if (txDetails.txData?.to?.value) {
+    parametersItems.push({
+      label: 'Contract',
+      render: () => (
+        <AddressDisplay
+          address={txDetails.txData?.to.value as Address}
+          copyProps={{ color: '$textSecondaryLight', size: 16 }}
+          externalLinkSize={16}
+        />
+      ),
+    })
+  }
+
+  // To (only show if different from Contract - in this case it's the same)
+  if (txDetails.txData?.to?.value) {
```

Comment:
The condition is the same as above though so we always have both.

Reply:
Yes, I'll leave just the "to" label.